replication/row_event.go

```diff
@@ -807,27 +811,44 @@ func (e *TableMapEvent) IsEnumOrSetColumn(i int) bool {
 	return rtyp == MYSQL_TYPE_ENUM || rtyp == MYSQL_TYPE_SET
 }
 
+// JsonColumnCount returns the number of JSON columns in this table
+func (e *TableMapEvent) JsonColumnCount() uint64 {
+	e.jsonColumnCountOnce.Do(func() {
```

Comment:
Just to confirm, this `sync.Once` wants to avoid duplicate work for binlog events pattern "TableMapEvent, RowsEvent, RowsEvent, RowsEvent..." (multiple RowsEvents share same TableMapEvent)? Because I see we always create a new TableMapEvent each time we see it in binlog

https://github.com/go-mysql-org/go-mysql/blob/e3201442601efb6995888212097f03d84bf993ca/replication/parser.go#L254-L257

Reply:
Ooops. I agree with this. For now, I'll just simplify the code by removing Once.

Reply:
updated